src/Core/Migration/V6_7/Migration1765287397AddConsentTable.php

```diff
@@ -0,0 +1,49 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Migration\V6_7;
+
+use Doctrine\DBAL\Connection;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\Framework\Migration\MigrationStep;
+
+/**
+ * @internal
+ */
+#[Package('framework')]
+class Migration1765287397AddConsentTable extends MigrationStep
+{
+    public function getCreationTimestamp(): int
+    {
+        return 1765287397;
+    }
+
+    public function update(Connection $connection): void
+    {
+        $connection->executeStatement('
+            CREATE TABLE IF NOT EXISTS `consent_state` (
+                `id` BINARY(16) NOT NULL,
+                `name` VARCHAR(100) NOT NULL,
+                `identifier` BINARY(16) NULL,
+                `state` VARCHAR(20) NOT NULL,
+                `actor_id` BINARY(16) NOT NULL,
```

Comment:
Is it possible to add a FK for this? I assume it should reference the user who initiatied this state? 
Thinking about this a small ADR might make sense to describe how the system is intended to work?

Reply:
Especially if we want to use the new consent stuff to handle the `requested_permissions` for apps.